File: libclang/src/main/java/klang/jvm/binding/CXIdxAttrInfo.java

package klang.jvm.binding;

import com.sun.jna.Pointer;
import com.sun.jna.Structure;
import org.jetbrains.annotations.NotNull;

@SuppressWarnings("unused")
@Structure.FieldOrder({"kind", "cursor", "loc"})
public class CXIdxAttrInfo extends Structure {
    public int kind;
    public CXCursor.CXCursorByValue cursor;
    public CXIdxLoc.ByValue loc;

    public CXIdxAttrInfo(@NotNull Pointer pointer) {
        super(pointer);
        read();
    }

}

File: libclang/src/main/java/klang/jvm/binding/CXIdxAttrInfo.kt
package klang.jvm.binding

import com.sun.jna.Pointer
import com.sun.jna.Structure
import klang.jvm.binding.CXCursor.CXCursorByValue

@Suppress("unused")
@Structure.FieldOrder("kind", "cursor", "loc")
class CXIdxAttrInfo(pointer: Pointer) : Structure(pointer) {

    var kind = 0
    var cursor = CXCursorByValue()
    var loc = CXIdxLoc.ByValue()

    init {
        read()
    }
}
